src/ray/core_worker/reference_count.h

```diff
@@ -37,6 +37,12 @@ class ReferenceCounter {
   /// Returns the total number of ObjectIDs currently in scope.
   size_t NumObjectIDsInScope() const LOCKS_EXCLUDED(mutex_);
 
+  /// Return whether this object has an active reference.
+  bool HasReference(const ObjectID &object_id) LOCKS_EXCLUDED(mutex_);
+
+  /// Sets or resets the callback to run when an object is deleted.
+  void OnObjectDeleted(std::function<void(const ObjectID &)> callback);
```

Comment:
Can we just pass this into the constructor? It seems like the CoreWorker could set the callback when constructing ref_counter_.

Reply:
Not easily, since it's a circular dependency. I think it's more straightforward to mutate the class.

Reply:
Can't you pass in `[this](...) { memory_store_.Delete(...) }` from the core worker constructor?

Reply:
I thought about that, but it seems more subtle to me. The current approach is straightforward, works, and is already implemented.